qadence_protocols/measurements/abstract.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+from qadence import QuantumModel
+from qadence.blocks.abstract import AbstractBlock
+from torch import Tensor
+
+from qadence_protocols.types import MeasurementData
+
+
+class MeasurementManager(ABC):
+    """The abstract class that defines the interface for the managing measurements.
+
+    Attributes:
+        measurement_data (MeasurementData, optional): Measurement data if already obtained.
+        options (dict, optional): Dictionary of options specific to protocol.
+    """
+
+    def __init__(self, measurement_data: MeasurementData = None, options: dict = dict()):
```

Comment:
```suggestion
    def __init__(self, measurement_data: MeasurementData | None = None, options: dict = dict()):
```

Unless `None` makes sense wrt `MeasurementData` type.

Reply:
For me None refers to no data obtained yet

Reply:
Ah ok, you mean decouple the None from measurement data